spec/support/shared_examples/create_service_with_price_rule_shared_examples.rb

```diff
@@ -0,0 +1,21 @@
+RSpec.shared_examples "creates a product with billing mode" do |product_type, billing_mode|
```

Comment:
nit: please add `# frozen_string_literal: true`

nit 2: please add newline at the end of the file

nit 3: the filename should be something like "create_product_with_price_rule_shared_examples.rb", right? We're not only creating services